src/redux/actions/createPost.ts

```diff
@@ -0,0 +1,50 @@
+import {Dispatch} from "redux";
+import {getToken} from "../../utils/getToken";
+import axios, {AxiosError} from "axios";
+import {URI} from "../../URI";
+import {User} from "../reducers/User";
+import {createAction} from "@reduxjs/toolkit";
+import {
+    CREATE_POST_FAILED,
+    CREATE_POST_REQUEST,
+    CREATE_POST_SUCCESS,
+} from "../actionTypes/actionTypes";
+
+const createPostRequest = createAction(CREATE_POST_REQUEST);
+const createPostSuccess = createAction<any, typeof CREATE_POST_SUCCESS>(CREATE_POST_SUCCESS);
+const createPostFailed = createAction<any, typeof CREATE_POST_FAILED>(CREATE_POST_FAILED);
+
+export type CreatePostsAction =
+    | ReturnType<typeof createPostRequest>
+    | ReturnType<typeof createPostSuccess>
+    | ReturnType<typeof createPostFailed>;
+
+export const createPostAction =
+    (
+        title: string,
+        image: string,
+        user: User,
+        replies: Array<{ title: string; image: string; user: User }>,
+    ) =>
+        async (dispatch: Dispatch<CreatePostsAction>) => {
+            try {
+                dispatch(createPostRequest());
+
+                const token = await getToken();
+
+                const {data} = await axios.post(
+                    `${URI}/create-post`,
+                    {title, image, user, replies},
+                    {
+                        headers: {Authorization: `Bearer ${token}`},
+                    });
```

Comment:
Until it's too late let's work on our network layer. We have the "get token and attach it to request" logic repetitive. What's worse, we don't have proper auth error and retry handling in place. 

That's definitely should not be a part of this PR, let's schedule it for the next one. 

We need:
1) Common interface for authorized requests. Something like "authGet", or "authFetch". The API is up to you to decide
2) Common token expiration validations
3) Common handling for unauthorized errors. We may want to try getting new tokens when we get a 401 and log the user out if unsuccessful